app/controllers/payments_controller.rb

```diff
@@ -0,0 +1,58 @@
+class PaymentsController < ApplicationController
+  before_action :set_category, only: %i[index new create]
+  before_action :set_payment, only: [:destroy]
+
+  def index
+    @payments = @category.payments.order(id: :desc)
+  end
+
+  def new
+    @payment = Payment.new
+  end
+
+  def create
+    @payment = current_user.payments.build(payment_params)
+    @categories = Category.where(id: payment_params[:category_ids])
+
+    if @payment.save
+      @categories.each { |category| category.payments << @payment unless category.payments.include?(@payment) }
+      flash[:success] = 'Transaction was created!'
+      redirect_to category_payments_path(@category)
+    else
```

Comment:
[Optional] but very important , the aproach you are following is used when you have a many to many relation without a model but since you have already created the CategoryPayment model to the payment category relation then you can just use this 
` if @payment.save
    @categories.each do |category|
      CategoryPayment.create(category: category, payment: @payment)
    end
`
then has_many :payments, through: :category_payments  will be enough to retreive related items